react-state/hannah-app/src/ToggleButton.jsx

```diff
@@ -3,25 +3,14 @@ import { useState } from 'react';
 export default function ToggleButton({ text, color }) {
   const [isClicked, setClick] = useState(false);
   console.log('useState ', isClicked);
-  const buttonStyle = {
-    backgroundColor: color,
-  };
 
   function handleClick() {
     console.log('before setter ', isClicked);
     setClick(!isClicked);
     console.log('after setter ', isClicked);
   }
   return (
-    <button
-      onClick={handleClick}
-      style={
-        isClicked
-          ? buttonStyle
-          : {
-              backgroundColor: color,
-            }
-      }>
+    <button onClick={handleClick} style={isClicked ? color : 'white'}>
```

Comment:
This is better, but it still isn't quite right. There are two cases here that your code will produce, one where `isClicked` is true and one where it is false.

If `isClicked` is true, you will render
`style={color}`
if false, you will render
`style={'white'}`.

But what does `style` do on a button element? Its value needs to be an object with properties indicating which CSS style will be applied. Here, you are setting it to just a string, not an object.

Your code should be something more like, `style={{ backgroundColor: isClicked ? clickedColor : unclickedColor }}`

Parsing this code, you read it as "style is being set to whatever value the expression inside the `{}` evaluates to; that value is the object `{ backgroundColor: isClicked ? clickedColor : unclickedColor }`". Parsing that even further, you see that it is an object with one property (`backgroundColor`) whose value depends on whether `isClicked` is true or false.

Reply:
Thank you for the explanation!!